### ISSUE TITLE
pty_fork: 24 (Too many open files) and cause infinite loop

### ISSUE THREAD

**Issue by futurist** [tsl0922/ttyd#926]
**Describe the bug**

Don't know when and why to cause this problem but when realized the below infinite loop shown.

```
[2022/05/17 14:31:02:1068] E: pty_fork: 24 (Too many open files)
[2022/05/17 14:31:02:1068] N: WS closed from 127.0.0.1, clients: 0
[2022/05/17 14:31:02:1069] N: __lws_lc_untag:  -- [wsisrv|8ef|adopted] (0) 2.833ms
[2022/05/17 14:31:02:1515] N: __lws_lc_tag:  ++ [wsisrv|8f0|adopted] (1)
[2022/05/17 14:31:02:1516] N: HTTP / - 127.0.0.1
[2022/05/17 14:31:02:1524] N: __lws_lc_untag:  -- [wsisrv|8f0|adopted] (0) 916μs
[2022/05/17 14:31:02:1558] N: __lws_lc_tag:  ++ [wsisrv|8f1|adopted] (1)
[2022/05/17 14:31:02:1559] N: WS   /ws - 127.0.0.1, clients: 1
[2022/05/17 14:31:02:1583] E: pty_fork: 24 (Too many open files)
[2022/05/17 14:31:02:1584] N: WS closed from 127.0.0.1, clients: 0
[2022/05/17 14:31:02:1584] N: __lws_lc_untag:  -- [wsisrv|8f1|adopted] (0) 2.661ms
[2022/05/17 14:31:02:1975] N: __lws_lc_tag:  ++ [wsisrv|8f2|adopted] (1)
[2022/05/17 14:31:02:1976] N: HTTP / - 127.0.0.1
[2022/05/17 14:31:02:1989] N: __lws_lc_untag:  -- [wsisrv|8f2|adopted] (0) 1.374ms
[2022/05/17 14:31:02:2063] N: __lws_lc_tag:  ++ [wsisrv|8f3|adopted] (1)
[2022/05/17 14:31:02:2066] N: WS   /ws - 127.0.0.1, clients: 1
... ...
```

The only way to close is press `Ctrl+C` and restart is OK.

**Expected behavior**
Should support long running in background without this issue.

**Screenshots**

![image](https://user-images.githubusercontent.com/159167/168744495-4bd1b009-dfbe-433e-8b79-a73eec2d77a5.png)

**Environment:**
 - OS: [macOS 10.15.7]
 - Browser: [Chrome 100.0.4896.60 (x86_64)]

**Additional context**

**Comment by tsl0922** [comment]
1. run `lsof | awk '{print $2}' | sort | uniq -c | sort -nr | head -n 10` to find the process ids that open many fd
2. run `ps -ef | grep <pid>` to find the process name

**Comment by angles-n-daemons** [comment]
I'm having this issue too - hoping to reproduce the above suggestion. I suppose it's probably chrome which is holding the open files, but there's no other issues on my computer - only ttyd fails in this way

**Comment by angles-n-daemons** [comment]
I ran the above command and it seems like Arc (Chromium based browser) and ttyd were holding a good portion of the connections.

```
➜  ~ lsof | awk '{print $2}' | sort | uniq -c | sort -nr | head -n 10
 469 62481
 279 369
 263 10819
 258 487
 191 477
 166 63913
 153 432
 145 77492
 134 413
 133 400
➜  ~ ps -e | grep 62481
62481 ??        15:14.56 /Applications/Arc.app/Contents/MacOS/Arc
66055 ttys094    0:00.00 grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn --exclude-dir=.idea --exclude-dir=.tox 62481
➜  ~ ps -e | grep 10819
10819 ttys001    0:21.59 ttyd -i 127.0.0.1 /bin/zsh
66062 ttys094    0:00.01 grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn --exclude-dir=.idea --exclude-dir=.tox 10819
➜  ~ ps -e 369
  PID TTY           TIME CMD
  369 ??         0:45.76 /usr/libexec/UserEventAgent (Aqua)
➜  ~ ps -e 487
  PID TTY           TIME CMD
  487 ??        34:10.80 /System/Library/Frameworks/CoreServices.framework/Frameworks/Metadata.framework/Versions/A/Support/corespotlightd
```

**Comment by angles-n-daemons** [comment]
Though they were holding a lot, I'm not sure that the system was flooded. I was again seeing the error mentioned:

```
(libuv) kqueue(): Too many open files
[2023/08/31 16:02:32:3106] E: pty_spawn: 24 (Too many open files)
[2023/08/31 16:02:32:3108] N: WS closed from 127.0.0.1, clients: 0
[2023/08/31 16:02:32:3109] N: __lws_lc_untag:  -- [wsisrv|4bb|adopted] (527) 3.133ms
[2023/08/31 16:02:32:3133] N: __lws_lc_tag:  ++ [wsisrv|4bc|adopted] (528)
[2023/08/31 16:02:32:3133] E: [wsisrv|4bc|adopted]: elops_accept_uv: uv_poll_init failed -17, sockfd=0x0
[2023/08/31 16:02:32:3285] N: __lws_lc_tag:  ++ [wsisrv|4bd|adopted] (529)
[2023/08/31 16:02:32:3291] N: WS   /ws - 127.0.0.1, clients: 1
(libuv) kqueue(): Too many open files
[2023/08/31 16:02:32:3335] E: pty_spawn: 24 (Too many open files)
[2023/08/31 16:02:32:3338] N: WS closed from 127.0.0.1, clients: 0
[2023/08/31 16:02:32:3339] N: __lws_lc_untag:  -- [wsisrv|4bd|adopted] (528) 5.333ms
[2023/08/31 16:02:32:3375] N: __lws_lc_tag:  ++ [wsisrv|4be|adopted] (529)
[2023/08/31 16:02:32:3376] E: [wsisrv|4be|adopted]: elops_accept_uv: uv_poll_init failed -17, sockfd=0x0
[2023/08/31 16:02:32:3399] N: __lws_lc_tag:  ++ [wsisrv|4bf|adopted] (530)
[2023/08/31 16:02:32:3402] N: WS   /ws - 127.0.0.1, clients: 1
```
